File: app/src/main/java/com/example/macarus0/walkiewalkie/data/WalkWithDogsDao.java
package com.example.macarus0.walkiewalkie.data;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

@Dao
public abstract class WalkWithDogsDao {
    @Query("SELECT * from Dog JOIN walkwithdogs on Dog.dogId = walkWithDogs.dogId " +
            "WHERE walkId = :walkId")
    public abstract List<Dog> getDogsOnWalk(long walkId);

    @Insert
    public abstract void insert(List<WalkWithDogs> walkWithDogs);
}

File: app/src/main/java/com/example/macarus0/walkiewalkie/data/WalkWithDogsDao.kt
package com.example.macarus0.walkiewalkie.data

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query

@Dao
interface  WalkWithDogsDao {
    @Query("SELECT * from Dog JOIN walkwithdogs on Dog.dogId = walkWithDogs.dogId " + "WHERE walkId = :walkId")
    fun getDogsOnWalk(walkId: Long): List<Dog>

    @Insert
    fun insert(walkWithDogs: List<WalkWithDogs>)

    @Query("DELETE FROM walkwithdogs WHERE walkwithdogs.dogId = :dogId")
    fun deleteDog(dogId: Long)
}
